velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -19,6 +19,7 @@
 #include "velox/connectors/hive/iceberg/IcebergDeleteFile.h"
 #include "velox/connectors/hive/iceberg/IcebergMetadataColumns.h"
 #include "velox/connectors/hive/iceberg/IcebergSplit.h"
+#include "velox/dwio/dwrf/writer/FlushPolicy.h"
```

Comment:
Why is it still dependent on DWRF? Can you please remove the dependency? You may need to do some more refactoring to use the dwio/common/ flush policy.

Reply:
There's this piece of code:
```
HiveIcebergTest()
      : config_{std::make_shared<facebook::velox::dwrf::Config>()} {
    // Make the writers flush per batch so that we can create non-aligned
    // RowGroups between the base data files and delete files
    flushPolicyFactory_ = []() {
      return std::make_unique<dwrf::LambdaFlushPolicy>([]() { return true; });
    };
  }
  ```
If I remove `#include "velox/dwio/dwrf/writer/FlushPolicy.h"` then it won't work...

Reply:
You can argue that the `LambdaFlushPolicy` should be a format independent `FlushPolicy` implementation and be moved to dwio::common. `FlushPolicy` is an interface already in common.

You can do this as a separate PR. Check if the Parquet also implements the same LambdaFlushPolicy. The thing about it is that nothing from the format is taken into account. Only the lambda decides so it is common among the formats.

Reply:
Looking at DWRF LambdaFlushPolicy in dwrf/FlushPolicy.h, it inherits a function from the DWRFFlushPolicy class called shouldFlushDictionary(). 

Also looking at Parquet LambdaFlushPolicy in parquet/writer/Writer.h, it inherits 2 variables from Parquet's DefaultFlushPolicy class that are not in the common FlushPolicy:
```
uint64_t rowsInRowGroup,
int64_t bytesInRowGroup,
```

Looks like since LambdaFlushPolicy inherits some properties/functions from a FlushPolicy class that's specifically for its respective file formats (dwrf and parquet), we shouldn't move the LambdaFlushPolicy to be in the common FlushPolicy.h... @czentgr What do you think?

Reply:
I see, the FlushPolicy's are format specific for certain function calls. And if they derive from a different class then the usage wouldn't work anymore. You would need to build a class that inherits from both, the common LambdaFlushPolicy and the format specific FlushPolicy. But this type of polymorphic inheritance might not be desired (could get messy). Doesn't look like there is an easy solution to this unless the common LambdaFlushPolicy combines the format specific API/members.

Reply:
@yingsu00 Can we maybe get your thoughts on how we can refactor accordingly for this?

Reply:
@minhancao 
I think we can just use the following structure:
```
             dwio::common::FlushPolicy
            /                             \
DWRFFlushPolicy          ParquetFlushPolicy (original DefaultFlushPolicy)
          |                                       |
dwrf::LambdaFlushPolicy            parquet::LambdaFlushPolicy
```

Reply:
https://github.com/facebookincubator/velox/pull/8864

Reply:
@minhancao So these recent changes are already made by someone else and would save you some effort. Then why is this file still dependent  on dwrf? Please, let it depend on dwio::common, not dwrf.

Reply:
@yingsu00 
I don't think it can depend on dwio::common because LambdaFlushPolicy is either respectively in DWRF or Parquet's flush policy. LambdaFlushPolicy is not in dwio::common::FlushPolicy.

Also, the flow now that I am implementing is you instantiate the respective file format's writer options in the test and then pass it to HiveConnectorTestBase::writeToFile() function where there will be dynamic casting of the WriterOptions based on the FileFormat passed into the function.

This PR's implementation is based on this PR https://github.com/facebookincubator/velox/pull/10915 that was merged in last month.